app/apptesting/test_suite.go

```diff
@@ -0,0 +1,249 @@
+// of one consensus engine unit in the default token of the app from first genesis
+// account. A Nop logger is set in app.
+func SetupWithGenesisValSet(t *testing.T, valSet *tmtypes.ValidatorSet, genAccs []authtypes.GenesisAccount, balances ...banktypes.Balance) *app.App {
+	t.Helper()
+
+	cudosApp, genesisState := setup(true, 5)
+	genesisState = genesisStateWithValSet(t, cudosApp, genesisState, valSet, genAccs, balances...)
+
+	stateBytes, err := json.MarshalIndent(genesisState, "", " ")
+	require.NoError(t, err)
+
+	// init chain will set the validator set and initialize the genesis accounts
+	cudosApp.InitChain(
+		abci.RequestInitChain{
+			ChainId:         SimAppChainID,
+			Validators:      []abci.ValidatorUpdate{},
+			ConsensusParams: DefaultConsensusParams,
+			AppStateBytes:   stateBytes,
+		},
+	)
+
+	// commit genesis changes
+	cudosApp.Commit()
+	cudosApp.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{
+		ChainID:            SimAppChainID,
+		Height:             cudosApp.LastBlockHeight() + 1,
+		AppHash:            cudosApp.LastCommitID().Hash,
+		ValidatorsHash:     valSet.Hash(),
+		NextValidatorsHash: valSet.Hash(),
+	}})
+
+	return cudosApp
+}
+
+func setup(withGenesis bool, invCheckPeriod uint) (*app.App, app.GenesisState) {
+	db := dbm.NewMemDB()
+	encCdc := app.MakeEncodingConfig()
+
+	cudosApp := app.New(
+		log.NewNopLogger(),
+		db,
+		nil,
+		true,
+		map[int64]bool{},
+		app.DefaultNodeHome,
+		invCheckPeriod,
+		encCdc,
+		simapp.EmptyAppOptions{},
+	)
+	if withGenesis {
+		return cudosApp, app.NewDefaultGenesisState(encCdc.Codec)
+	}
+
+	return cudosApp, app.GenesisState{}
+}
+
+func genesisStateWithValSet(t *testing.T,
+	app *app.App, genesisState app.GenesisState,
+	valSet *tmtypes.ValidatorSet, genAccs []authtypes.GenesisAccount,
+	balances ...banktypes.Balance,
+) app.GenesisState {
+	// set genesis accounts
+	authGenesis := authtypes.NewGenesisState(authtypes.DefaultParams(), genAccs)
+	genesisState[authtypes.ModuleName] = app.AppCodec().MustMarshalJSON(authGenesis)
+
+	validators := make([]stakingtypes.Validator, 0, len(valSet.Validators))
+	delegations := make([]stakingtypes.Delegation, 0, len(valSet.Validators))
+
+	bondAmt := sdk.DefaultPowerReduction
+
+	for _, val := range valSet.Validators {
+		pk, err := cryptocodec.FromTmPubKeyInterface(val.PubKey)
+		require.NoError(t, err)
+		pkAny, err := codectypes.NewAnyWithValue(pk)
+		require.NoError(t, err)
+		validator := stakingtypes.Validator{
+			OperatorAddress:   sdk.ValAddress(val.Address).String(),
+			ConsensusPubkey:   pkAny,
+			Jailed:            false,
+			Status:            stakingtypes.Bonded,
+			Tokens:            bondAmt,
+			DelegatorShares:   sdk.OneDec(),
+			Description:       stakingtypes.Description{},
+			UnbondingHeight:   int64(0),
+			UnbondingTime:     time.Unix(0, 0).UTC(),
+			Commission:        stakingtypes.NewCommission(sdk.ZeroDec(), sdk.ZeroDec(), sdk.ZeroDec()),
+			MinSelfDelegation: sdk.ZeroInt(),
```

Comment:
@dao-notional I think the MinSelfDelegation should be set here, it is not supposed to be zero.

Also, why the x/bank/keeper tests still pass when the MinSelfDelegaton is being set to zero. Isn't it supposed to be caught by the newly added ante?